src/testing/mocks.rs

```diff
@@ -343,3 +347,19 @@ impl NodeImplementation<MockTypes> for MockNodeImpl {
         (ChannelMaps::new(start_view), None)
     }
 }
+
+#[async_trait]
+pub trait TestableDataSource:
+    AvailabilityDataSource<MockTypes, MockNodeImpl>
+    + StatusDataSource
+    + UpdateDataSource<MockTypes, MockNodeImpl>
+    + Send
+    + Sync
+    + Sized
+    + 'static
+{
+    type TmpData;
+
+    async fn create(node_id: usize) -> (Self, Self::TmpData);
+    async fn commit_version(&mut self);
```

Comment:
May be out of scope - but would it be helpful for this to return a version number?

Reply:
I'm not currently aware of a use case for this